codex-ui/src/vue/components/select/Select.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <Button
+    :icon="activeItem.icon"
+    trailing-icon="BracketsVertical"
+    secondary
+    @click="togglePopover($event.currentTarget, {vertically: 'below', horizontally: 'left'})"
+  >
+    {{ activeItem.title }}
+  </Button>
+</template>
+<script setup lang="ts">
+import type { ContextMenuItem as SelectItem, DefaultItem } from '../context-menu/ContextMenu.types';
+import { ContextMenu } from '../context-menu';
+import { onMounted, ref } from 'vue';
+import { usePopover, PopoverShowParams } from '../popover';
+import { Button } from '../button';
+
+const props = defineProps<{
+  items: SelectItem[];
+  defaultItem: DefaultItem;
+}>();
+
+const items = props.items;
+const { showPopover, hide, isOpen } = usePopover();
+
+const togglePopover = (el: HTMLElement, align: PopoverShowParams['align']) => {
+  if (!isOpen.value) {
+    showPopover({
+      targetEl: el,
+      with: {
+        component: ContextMenu,
+        props: {
+          items: items,
+        },
+      },
+      align,
```

Comment:
lets also pass `width: 'fit-target'` property to make popover width same as  a button width